modules/standard/CPtr.chpl

```diff
@@ -337,30 +337,53 @@ module CPtr {
     return __primitive("cast", t, x);
   }
 
+  // casts from c pointer to c_intptr / c_uintptr
   pragma "no doc"
   inline operator :(x:c_void_ptr, type t:c_intptr)
     return __primitive("cast", t, x);
   pragma "no doc"
   inline operator :(x:c_void_ptr, type t:c_uintptr)
     return __primitive("cast", t, x);
+
   pragma "no doc"
-  inline operator :(x:c_void_ptr, type t:int(64)) where c_uintptr != int(64)
+  inline operator :(x:c_ptr, type t:c_intptr)
     return __primitive("cast", t, x);
   pragma "no doc"
-  inline operator :(x:c_void_ptr, type t:uint(64)) where c_uintptr != uint(64)
+  inline operator :(x:c_ptr, type t:c_uintptr)
```

Comment:
Since this has been expanded from `c_void_ptr` to `c_ptr(t)`, I think tests of these cases should be added?  Or are these things that we have existing tests for, which were relying on c_ptr->c_void_ptr coercions?

Reply:
(or wait... are these the same casts as before, just in a different order?)

Reply:
Yes, I just reordered them to make the code more understandable (IMO).